### PR TITLE
Update gradle plugin to support modern android gradle plugin

### PR DESCRIPTION
**Pull request by loopj**

### CODE REVIEW

**Comment by loopj** [comment]
@kattrali this is ready for review if you have time!

**Comment by loopj** [comment]
@kattrali also please feel free to be brutal - my groovy skills are limited :D

File: README.md [hunk 1]
```diff
@@ -45,6 +45,15 @@ bugsnag {
...
     // are using Bugsnag Enterprise, you can configure your upload endpoint
     endpoint 'http://upload.bugsnag.example.com'
+
+    // By default, we'll upload mapping files automatically, if you'd prefer
+    // to upload proguard mappings manually, you can set this to `false`
+    autoUpload false
+
+    // By default, we'll automatically apply our recommended proguard
+    // configuration to your existing proguard settings, if you'd prefer
+    // to do this yourself, you can set this to `false`
```
Comment (snmaynard) [inline README.md]: Is it worth explaining what this is? Seems a little scary/vague as is. Might be documented elsewhere though

Comment (loopj) [inline README.md]: Updated README to clarify

**Comment by kattrali** [comment]
@loopj This looks good. If you've got them handy, it would be good to document any other gradle plugins/repositories referenced for next time anyone needs to go down the rabbit hole.

**Pull request merged**